packages/dds/merge-tree/src/attributionPolicy.ts

```diff
@@ -188,6 +196,50 @@ function createPropertyTrackingMergeTreeCallbacks(
 	};
 }
 
+const customAttributeOnSegments = (
+	deltaSegments: IMergeTreeSegmentDelta[],
+	{ op }: IMergeTreeDeltaOpArgs,
+): void => {
+	const propName = customAttributionKeysPropName;
+	for (const { segment } of deltaSegments) {
+		const shouldAttributeInsert =
+			op.type === MergeTreeDeltaType.INSERT && segment.properties?.[propName] !== undefined;
+
+		if (shouldAttributeInsert && isCustomAttributionKeyList(segment.properties?.[propName])) {
+			const customAttributionKeyList = segment.properties?.[
+				propName
+			] as ICustomAttributionKeyList;
+			const customCollection = new CustomAttributionCollection(
+				segment.cachedLength,
+				customAttributionKeyList,
+			);
+			segment.attribution?.update(customAttributionKeysPropName, customCollection);
+			// Delete it from the segment properties as we don't want it to get summarized twice as
+			// we have added it in segment.attribution which will be summarized.
+			delete segment.properties?.[propName];
```

Comment:
delete is rather expensive in JS (usually). is it possible to use segment.properties?.[propName] = undefined instead?
It's not the same as delete.

Some quick search results (if you are curious to read more): https://stackoverflow.com/questions/59431589/javascript-performance-constraints-of-delete-keyword